### PR TITLE
fix(#8252): Adds retry when getting userCtx when logging in

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

In clustered couchdb, if a user's password is updated on one node, and immediately afterwards another node receives a login request, there's a chance it will respond with a 401 because the nodes are not in sync. Adding a retry until the request either hits the same node, or enough time has passed that the nodes are already in sync, 

medic/cht-core#8252

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8252-sync-shards-after-changing-password/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8252-sync-shards-after-changing-password/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8252-sync-shards-after-changing-password/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Hi @jkuester 
Would you mind reviewing this tiny PR? 
Thank you!

File: api/src/controllers/login.js [hunk 1]
```diff
@@ -204,6 +201,18 @@ const renderTokenLogin = (req, res) => {
     .then(body => res.send(body));
 };
 
+const getUserCtxRetry = async (options, retry = 10) => {
+  try {
+    return await auth.getUserCtx(options);
+  } catch (err) {
+    if (retry > 0) {
```
Comment (jkuester) [inline api/src/controllers/login.js:208]: Should we check if the error was a `404` before retrying?  (Also is it a `404` error or `401` that we would get?  PR description says `404`...)

Also, can you add a brief in-line comment here to explain why we think we might get a different response when retrying?

**Review by jkuester — APPROVED** [review]
:tada:

File: api/src/controllers/login.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dianabarsan) [reply api/src/controllers/login.js:208]: I've updated the description, it's supposed to be a 401. I'll add the check and the comment.

**Pull request merged**